build/Build.Steps.cs

```diff
@@ -486,43 +485,46 @@ partial class Build
 
                         CopyNativeDependenciesToStore(file, jsonDocument, architectureStores);
 
-                        depsJsonContent = RemoveOpenTelemetryAutoInstrumentationAdditionalDepsFromDepsContent(depsJsonContent);
+                        RemoveOpenTelemetryAutoInstrumentationAdditionalDepsFromDepsContent(jsonDocument);
```

Comment:
nit: since the `jsonDocument` is mutable, it may work better as an extension method (possibly paired with a rename for the method)